### PR TITLE
Document the `optional` method

### PR DESCRIPTION
**Pull request by sgrif**
I think this was the first time I've written a doctest for Diesel which
required 0 setup. \o/

### CODE REVIEW

File: diesel/src/result.rs [hunk 1]
```diff
@@ -82,7 +82,26 @@ pub enum ConnectionError {
 pub type QueryResult<T> = Result<T, Error>;
 pub type ConnectionResult<T> = Result<T, ConnectionError>;
 
+/// See the [method documentation](#tymethod.optional).
```
Comment (killercup) [inline diesel/src/result.rs]: This link depends on the way rustdoc adds ids to certain elements on the page. I don't think this is public API and will work in future version of rustdoc. But let's keep it for now and hopefully just upgrade to glorious intra-rustdoc links later!

File: diesel/src/result.rs [hunk 2]
```diff
@@ -82,7 +82,26 @@ pub enum ConnectionError {
...
 pub trait OptionalExtension<T> {
+    /// Converts a `QueryResult<T>` into a `QueryResult<Option<T>>`.
+    ///
+    /// By default, Diesel treats 0 rows being returned from a query that is
+    /// expected to return 1 row as an error. This method will handle that
```
Comment (killercup) [inline diesel/src/result.rs]: "…as constructed with `get_result`"

Comment (sgrif) [inline diesel/src/result.rs]: I'm not sure I understand where you want me to put that

Comment (killercup) [inline diesel/src/result.rs]: Sorry, more context:

> By default, Diesel treats 0 rows being returned from a query that expected to return 1 row (as e.g. constructed with `get_result`) as an error.

**Pull request merged**